tests/appsec/test_rate_limiter.py

```diff
@@ -37,22 +37,16 @@ def setup_main(self):
             self.requests.append(weblog.get("/waf/", headers={"User-Agent": "Arachni/v1"}))
             self.request_count += 1
 
-    @bug(context.library > "nodejs@3.14.1", reason="_sampling_priority_v1 is missing")
     def test_main(self):
         """send requests for 10 seconds, check that only 10-ish traces are sent, as rate limiter is set to 1/s"""
 
         MANUAL_KEEP = 2
         trace_count = 0
 
         for r in self.requests:
-            for data, _, span, _ in interfaces.library.get_appsec_events(request=r):
+            for _, _, span, _ in interfaces.library.get_appsec_events(request=r):
                 # the logic is to set MANUAL_KEEP not on all traces
                 # then the sampling mechism drop, or not the traces
-
-                assert (
-                    "_sampling_priority_v1" in span["metrics"]
-                ), f"_sampling_priority_v1 is missing in span {span['span_id']} in {data['log_filename']}"
-
                 if span["metrics"]["_sampling_priority_v1"] == MANUAL_KEEP:
```

Comment:
@cbeauchesne I don't know how to write that in Python, but the `span["metrics"]["_sampling_priority_v1"]` field is optional. So you need to gracefully ignore the trace if the field is not present.

Reply:
Hi @simon-id,

It's simply `"_sampling_priority_v1" in span["metrics"]`.

For this one, you can also do a one-liner with the `get` function of python `dict` that returns `None` if the field is missing : 

```
if span["metrics"].get("_sampling_priority_v1") == MANUAL_KEEP:
```